CMakeLists.txt

```diff
@@ -85,8 +85,23 @@ include(ROCMPackageConfigHelpers)
 include(ROCMInstallSymlinks)
 include(CheckCXXCompilerFlag)
 
+# check if asan is enabled
+if (NOT DEFINED ADDRESS_SANITIZER AND DEFINED ENV{ADDRESS_SANITIZER})
+    set(ADDRESS_SANITIZER $ENV{ADDRESS_SANITIZER})
+endif()
+if (ADDRESS_SANITIZER OR CMAKE_CXX_FLAGS MATCHES "-fsanitize=address")
+    set(ADDRESS_SANITIZER_ENABLED ON)
+else()
+    set(ADDRESS_SANITIZER_ENABLED OFF)
+endif()
+
+if (ADDRESS_SANITIZER_ENABLED)
+rocm_check_target_ids(DEFAULT_AMDGPU_TARGETS
```

Comment:
I think the tabbing is a bit off.
Also can remove the gfx908:xnack+ target  for xnack (don't think supported)